src/TheThingsNetwork.cpp

```diff
@@ -357,13 +357,18 @@ void TheThingsNetwork::clearReadBuffer()
   }
 }
 
-size_t TheThingsNetwork::readLine(char *buffer, size_t size)
+size_t TheThingsNetwork::readLine(char *buffer, size_t size, uint8_t attempts = 3) // Default timeout value 10s and is set in class initiator
 {
   size_t read = 0;
-  while (read == 0)
+  while (read == 0 && attempts>0)
```

Comment:
Spacing; `attemps > 0`

Reply:
Shouldn't you be decrementing `attempts` somewhere? Like `while (!read && attempts--)` ?